defdispjlib.py

```diff
@@ -87,24 +87,166 @@ def parse_deformable_displacement(rw, ed):
+                before re-importing the element.")
+        return {'CANCELLED'}
+
+    try:
+        n1 = nd['node_' + str(elem.nodes[0].int_label)].blender_object
+    except KeyError:
+        print("spawn_distance_element(): Could not find a Blender \
+                object associated to Node " + \
+              str(elem.nodes[0].int_label))
+        return {'NODE1_NOTFOUND'}
+
+    try:
+        n2 = nd['node_' + str(elem.nodes[1].int_label)].blender_object
+    except KeyError:
+        print("spawn_distance_element(): Could not find a Blender \
+                object associated to Node " + \
+              str(elem.nodes[1].int_label))
+        return {'NODE2_NOTFOUND'}
+
+    # nodes' objects
+    n1OBJ = bpy.data.objects[n1]
+    n2OBJ = bpy.data.objects[n2]
+
+    # creation of line representing the dist
+    defdispobj_id = 'dist_' + str(elem.int_label)
+    defdispcv_id = defdispobj_id + '_cvdata'
+
+    # check if the object is already present. If it is, remove it.
+    if defdispobj_id in bpy.data.objects.keys():
+        bpy.data.objects.remove(bpy.data.objects[defdispobj_id])
+
+    # check if the curve is already present. If it is, remove it.
+    if defdispcv_id in bpy.data.curves.keys():
+        bpy.data.curves.remove(bpy.data.curves[defdispcv_id])
+
+    # get offsets
+    f1 = elem.offsets[0].value
+    f2 = elem.offsets[1].value
+    q1 = elem.rotoffsets[0].value
+    q2 = elem.rotoffsets[1].value
+
+    # assign coordinates of knots in global frame
+    R1 = n1OBJ.rotation_quaternion.to_matrix()
+    R2 = n2OBJ.rotation_quaternion.to_matrix()
+    p1 = n1OBJ.location + R1 * Vector((f1[0], f1[1], f1[2]))
+    p2 = n2OBJ.location + R2 * Vector((f2[0], f2[1], f2[2]))
+
+    bpy.ops.object.empty_add(type='PLAIN_AXES')
+    defdispOBJ = bpy.context.scene.objects.active
+    defdispOBJ.location = p1
+    defdispOBJ.rotation_mode = 'QUATERNION'
+    defdispOBJ.rotation_quaternion = n1OBJ.rotation_quaternion * Quaternion(( q1[0], q1[1], q1[2], q1[3] ))
+
+    elem.blender_object = elem.name
+    defdispOBJ.name = elem.name
+    ude = bpy.context.scene.mbdyn.elems_to_update.add()
+    ude.dkey = elem.name
+    ude.name = elem.name
+
+    length = (p2 - p1)
+    length = Vector(map(abs, length))
+    radius = 0.3 * length
+    print(length, radius)
+    turns = 20
+    axes = ['X', 'Y', 'Z']
+    for ii in range(3):
+        rad_helix, len_helix = radius[ii], length[ii]
+        if rad_helix < 0.1:
+            rad_helix = 0.1
+            len_helix = rad_helix/ 0.3
+        bpy.ops.mesh.curveaceous_galore(ProfileType='Helix', helixHeight=len_helix, \
+                                        helixWidth=rad_helix, helixEnd=360* turns, helixPoints=1000)
+        defdispChild = bpy.context.scene.objects.active
+
+        track_axes = axes[:]
+        track_axes.remove(track_axes[ii])
+        obj_loc = (p2 - p1)
+        defdispChild.location = Vector(p1) + Vector(obj_loc[:ii] + (0, )* (3 - ii))
```

Comment:
Same thing as above: you are using the correct local components of the deformation vector, but you are applying them to the springs aligned along the global X, Y, Z axes.